packages/upload-media/README.md

```diff
@@ -0,0 +1,108 @@
+# `@wordpress/upload-media`
+
+Core media upload logic implemented with a custom `@wordpress/data` store.
+
+## API Reference
+
+### Actions
+
+The following set of dispatching action creators are available on the object returned by `wp.data.dispatch( 'core/upload-media' )`:
+
+<!-- START TOKEN(Autogenerated actions|src/store/actions.ts) -->
+
+#### addItems
+
+Adds a new item to the upload queue.
+
+_Parameters_
+
+-   _$0_ `AddItemsArgs`:
+-   _$0.files_ `AddItemsArgs[ 'files' ]`: Files
+-   _$0.onChange_ `[AddItemsArgs[ 'onChange' ]]`: Function called each time a file or a temporary representation of the file is available.
+-   _$0.onSuccess_ `[AddItemsArgs[ 'onSuccess' ]]`: Function called after the file is uploaded.
+-   _$0.onBatchSuccess_ `[AddItemsArgs[ 'onBatchSuccess' ]]`: Function called after a batch of files is uploaded.
+-   _$0.onError_ `[AddItemsArgs[ 'onError' ]]`: Function called when an error happens.
+-   _$0.additionalData_ `[AddItemsArgs[ 'additionalData' ]]`: Additional data to include in the request.
+-   _$0.allowedTypes_ `[AddItemsArgs[ 'allowedTypes' ]]`: Array with the types of media that can be uploaded, if unset all types are allowed.
+
+#### cancelItem
+
+Cancels an item in the queue based on an error.
```

Comment:
Based on this description, I wonder if this should be a public action. It sounds more as an action that the package uses internally rather than something the consumer does?

Reply:
The idea for making it public is that there can be a user-facing "uploads manager" in the block-editor UI (or anywhere else `upload-media` is used) that allows you to cancel uploads. For example if you accidentally dropped the wrong files or giant images that would take forever to upload.

In the media experiments plugin it looks like this:
![Screenshot 2024-12-12 at 15 32 45](https://github.com/user-attachments/assets/9bdc4f38-91f2-4fc3-8d01-9c1fcf57c616)